jf_agent/main.py

```diff
@@ -248,21 +256,24 @@ def obtain_creds(config):
 
     jira_username = os.environ.get('JIRA_USERNAME', None)
     jira_password = os.environ.get('JIRA_PASSWORD', None)
+    jira_bearer_token = os.environ.get('JIRA_BEARER_TOKEN', None)
 
     # obtain git slug to credentials
     git_instance_to_creds = {
         git_config.git_instance_slug: _get_git_instance_to_creds(git_config)
         for git_config in config.git_configs
     }
 
-    if config.jira_url and not (jira_username and jira_password):
+    # This is a bit hairy, they must either provide a jira username
+    # and a jira password OR a single jira bearer token.
+    if config.jira_url and ((not (jira_username and jira_password)) and (not jira_bearer_token)):
```

Comment:
`not ((jira_username and jira_password) ^ jira_bearer_token)` perhaps?

Reply:
This makes sense but also might be a bit less readable (unless this way is more pythonic?)